### ISSUE TITLE
Can't build Godot for Android (from Windows)

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot#6755]
Using Windows 10 64-bit with [this script](https://gist.github.com/Calinou/44edbd5d9c4a2ac0dbc8f165b8b7f0ce) to compile. MinGW-w64 is installed as it seems to be required for compiling for Android, and is in PATH. Android SDK 23.0.3 is used, with latest Android NDK.

```
Compiling shared ==> platform\android\os_android.cpp
=====
arm-linux-androideabi-g++: error: /Foplatform\android\os_android.os: No such file or directory
arm-linux-androideabi-g++: error: /c: No such file or directory
arm-linux-androideabi-g++: error: /TP: No such file or directory
```

Visual Studio 2015 is also installed, if that matters, but it is not in PATH for building for Android (it used to work just fine this way).

**Comment by SantiagoMenelao** [comment]
i have same issue in win7 
![image](https://cloud.githubusercontent.com/assets/14896681/19222513/af99cd10-8e30-11e6-874e-4715b03c9f38.png)

**Comment by akien-mga** [comment]
CC @RandomShaper and @volzhs.

**Comment by vnen** [comment]
@SantiagoMenelao don't use the Developer Command Prompt when building for Android. It confuses the buildsystem. Use a regular Command Prompt instead.

**Comment by RandomShaper** [comment]
It seems the build system is passing MSVC-specific flags (leading slash) 
to GCC so it's surely an environment detection issue.

**Comment by akien-mga** [comment]
Potential regression from #6501? Though IINM I did not backport it to the 2.1 branch.

**Comment by vnen** [comment]
> MinGW-w64 is installed as it seems to be required for compiling for Android

I don't think it's required for Android. It does try to detect, but this is because the detection for all platforms always run even if you select a specific one.

**Comment by volzhs** [comment]
I had similar issue and I solved it by using `cmd` as administrator permission, not using MSVC developer command prompt.

**Comment by vnen** [comment]
I just finished building for Android in the 2.1 branch. No problem for me. Using NDK r11c still though.

**Comment by TheoXD** [comment]
I had this issue once, don't remember what I did but it's not a bug. So don't label this as "bug" please.

**Comment by akien-mga** [comment]
Well it's a bug until proven otherwise. @Calinou builds templates every couple of days, so it's a regression in his case. Regressions are often bugs :p

**Comment by vnen** [comment]
Can confirm @Calinou's error on master.

**Comment by vnen** [comment]
Maybe it's from #6704? If the Windows detection is adding some flags, they might interfere.

**Comment by volzhs** [comment]
I didn't try it with recent over 20 merges.

**Comment by vnen** [comment]
I guess Akien was right, bisect points me to #6501.

**Comment by vnen** [comment]
Turns out to be a simple fix: #6757
